docs/madr/decisions/009-tracing-policy.md

```diff
@@ -0,0 +1,184 @@
+Backends can contain the same settings but they can be set or overriden in
+specific `Tracing` resources.
+
+#### `targetRef`
+
+This is a new policy so it's based on `targetRef`. Envoy tracing is configured
+on the HTTP connection manager so `Tracing` has a single `spec.targetRef` field
+that selects what it applies to. It does not use `to` or `from` fields.
+
+All logging configuration happens under `spec.default` so that users are able to
+override settings with more specific `targetRef`s.
+
+Resources supported by `spec.targetRef` are `Mesh`, `MeshSubset`, `MeshService`,
+`MeshServiceSubset`, `MeshGatewayRoute` and eventually the future
+evolution of `TrafficRoute`, which we'll call `MeshTrafficRoute` in this
+document.
+
+#### Policy-specific configuration
+
+This MADR proposes additional options be configurable on `Tracing` resources
+directly.
+
+##### Sampling
+
+At the moment, tracing backends only support Envoy's
+[`overall_sampling`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-field-extensions-filters-network-http-connection-manager-v3-httpconnectionmanager-tracing-overall-sampling)
+via the `sampling` field.
+
+This MADR proposes instead the following:
+
+```yaml
+spec:
+ default:
+  sampling:
+   overall: <percentage>
+   client: <percentage>
+   random: <percentage>
+```
+
+with the justification that it's not clear to users
+that when they set the simple `sampling` field we have now,
+whether traces started via `x-client-trace-id` are limited.
+This also enables disabling `x-client-trace-id`.
+
+##### Tags
+
+Envoy can set tags in traces using values from:
+
+- Literal value
+- Request header
+- Environment variable
+- Metadata
+
+We propose allowing users to configure custom tags using only literal values or
+request headers. Using Kuma leaves environment variables and Envoy metadata
+opaque for users so the use cases for configuring them (at least directly)
+in Kuma are limited.
+
+Tags can be configured as follows:
+
+```yaml
+spec:
+ default:
+  tags:
+   - name:
+     # exactly one of the following keys must be set
+     literal: ...
+     header:
+      name:
+      default:
+```
+
+##### Method/path specific
+
+Users can set the `spec.targetRef` to be a `MeshGatewayRoute` or
+`MeshTrafficRoute`.
+
+Both paths and methods are matchable in `MeshGatewayRoute` and we can ensure
+that `MeshTrafficRoute` also supports this matching.
+
+### Examples
+
+All examples assume Kubernetes.
+
+#### Simple
+
+This configures all `Dataplane` inbounds with `kuma.io/service: backend` to
+trace a maximum of 80% of traffic and adds the custom tag `team` with a value of
+`core` to spans.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Tracing
+metadata:
+ name: all
+ labels:
+  kuma.io/mesh: default
+spec:
+ targetRef:
+  kind: MeshService
+  name: backend
+ default:
+  backend: jaeger
```

Comment:
1. Backends are an [array right now](https://kuma.io/docs/1.8.x/policies/traffic-trace/#add-a-tracing-backend-to-the-mesh) so why are we limiting them to 1 value here?

2. In the MeshLogging MADR backend can be of type `Reference` or "inline" (meaning a concrete version of the backend) so I'd go with something like this for reference:

```yaml
backends:
  - reference:
      kind: TracingBackend
      name: jaeger
```

and for "inline" something like:

```yaml
backends:
  - zipkin:
      name: jaeger
      conf:
        url: http://jaeger-collector.mesh-observability:9411/api/v2/spans
      sampling...
```

would be good if you could add an example of these two usages.

Reply:
For logging, it makes sense to have multiple backends at the same time. For tracing, I don't think it makes sense. Also, it's not possible with Envoy right now.

Reply:
+1 to @slonka comment. Please provide an example with TracingBackend